File: backend/models/mesa.py
from database import db

class Mesa(db.Model):
    """Modelo de Mesa, representa uma mesa do estabelecimento."""
    __tablename__ = 'mesa'

    mesa_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    numero = db.Column(db.Integer, nullable=False, unique=True)
    capacidade = db.Column(db.Integer, nullable=False)
    status = db.Column(db.String(50), nullable=False, default='livre')

    def __repr__(self):
        """Retorna representação legível da mesa."""
        return f'<Mesa {self.numero}>'

    def to_dict(self):
        """Converte a mesa para dicionário serializável."""
        return {
            'mesa_id': self.mesa_id,
            'numero': self.numero,
            'capacidade': self.capacidade,
            'status': self.status
        } 

File: ComandasApp/src/ComandasApp.Core/Entities/Mesa.cs
using System.ComponentModel.DataAnnotations;

namespace ComandasApp.Core.Entities;

public class Mesa
{
    [Key]
    public int MesaId { get; set; }
    
    [Required]
    public int Numero { get; set; }
    
    [Required]
    public int Capacidade { get; set; }
    
    [Required]
    [StringLength(50)]
    public string Status { get; set; } = "livre";
    
    public override string ToString()
    {
        return $"Mesa {Numero}";
    }
}
